util/freebsd_install.sh

```diff
@@ -1,21 +1,32 @@
 #!/bin/sh
 util_dir=$(dirname "$0")
-pkg update
-pkg install -y \
-	git \
-	wget \
-	gmake \
-	gcc \
-	zip \
-	unzip \
-	avr-binutils \
-	avr-gcc \
-	avr-libc \
-	dfu-programmer \
-	dfu-util \
-	arm-none-eabi-gcc \
-	arm-none-eabi-binutils \
-	arm-none-eabi-newlib \
-	diffutils \
-	python3
-pip3 install -r ${util_dir}/../requirements.txt
+if [ $(id -u) = 0 ]
+then
+	pkg update
+	pkg install -y \
+		git \
+		wget \
+		gmake \
+		gcc \
+		zip \
+		unzip \
+		avr-binutils \
+		avr-gcc \
+		avr-libc \
+		dfu-programmer \
+		dfu-util \
+		avrdude \
+		arm-none-eabi-gcc \
+		arm-none-eabi-binutils \
+		arm-none-eabi-newlib \
+		diffutils \
+		python3
+	echo ""
+	echo "Re-run the setup as your normal user to install the qmk python dependencies"
+	exit 1
+else
+	echo "Make sure you run setup as root first to install base OS dependencies..."
+	echo ""
+
+	python3 -m pip install --user -r ${util_dir}/../requirements.txt
+fi
```

Comment:
If I understand the situation here, the user doesn't actually get the requirements unless they run this twice, once as root and once as themselves? That seems inefficient.

Reply:
This will install the requirements for only the user running the command and not globally for every user on the system.

Reply:
Yeah, latest packaging of `pip` in FreeBSD *strongly* encourages doing user installs to avoid conflicts and problems with `pkg`d versions of Python packages.

So... This one script wants to do stuff with global packages and user scoped items. We could improve further by invoking `sudo` for the `pkg` calls if `sudo` is installed.

Reply:
In #8783 I'm also recommending this for Linux.

Reply:
I've just pushed 6857fb9b5 that updates the script to invoke w/ `sudo` if it is available on the system, making this a one-time run in that scenario. `sudo` isn't part of base FreeBSD system, so the script should handle both possibilities.